staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go

```diff
@@ -217,8 +219,7 @@ func (dm *discoveryManager) fetchFreshDiscoveryForService(gv metav1.GroupVersion
 	writer := newInMemoryResponseWriter()
 	handler.ServeHTTP(writer, req)
 
-	switch writer.respCode {
```

Comment:
I think a switch might still be more readable here

```go
switch {
case writer.respCode == http.StatusNotModified:
  // ...cache case

case writer.respCode == http.StatusServiceUnavailable:
  // ...server unavailable

case writer.respCode == http.StatusOK && discovery.ContentTypeIsGVK(writer.Header().Get("Content-Type"), ...):
  // ...aggregated

default:
  // ...fallback to legacy
}
```

Reply:
Since `discovery.ContentTypeIsGVK(...)` returns two values, the condition would need to be checked outside the case statement. Eg:
```
isGVK, err := discovery.ContentTypeIsGVK(...)
switch {
case ...
case writer.respCode == http.StatusOK && isGVK && err == nil:
// ...aggregated
...
}
```

Wouldn't an if statement be slightly cleaner in this regard?

Reply:
heh, true

Reply:
Changed to a case statement with the GVK check outside of it, lmk if this is readable.